src/containers/AdminCampaignEdit/components/ApproveCampaignButton.tsx

```diff
@@ -0,0 +1,66 @@
+import Button from "@material-ui/core/Button";
+import Tooltip from "@material-ui/core/Tooltip";
+import React, { useCallback } from "react";
+
+import {
+  useGetCampaignStatusQuery,
+  useSetCampaignApprovedMutation
+} from "../../../../libs/spoke-codegen/src";
+import { useSpokeContext } from "../../../client/spoke-context";
+import { useAuthzContext } from "../../../components/AuthzProvider";
+
+export interface ApproveCampaignButtonProps {
+  campaignId: string;
+}
+
+export const ApproveCampaignButton: React.FC<ApproveCampaignButtonProps> = (
+  props
+) => {
+  const { campaignId } = props;
+  const { isSuperadmin } = useAuthzContext();
+  const { orgSettings } = useSpokeContext();
+  const { data, loading } = useGetCampaignStatusQuery({
+    variables: { campaignId }
+  });
+  const isApproved = data?.campaign?.isApproved ?? false;
+  const [
+    setCampaignApproved,
+    { loading: approving }
+  ] = useSetCampaignApprovedMutation();
+
+  const working = loading || approving;
+
+  const tooltipText =
+    !isSuperadmin && orgSettings?.startCampaignRequiresApproval
+      ? "Superadmin approval required"
+      : "";
+
+  const handleClick = useCallback(() => {
+    if (!isSuperadmin || working) return;
+
+    setCampaignApproved({
+      variables: { campaignId, approved: !isApproved }
+    });
+  }, [campaignId, isSuperadmin, working, isApproved, setCampaignApproved]);
+
+  if (!isSuperadmin || !orgSettings?.startCampaignRequiresApproval) {
```

Comment:
Suggestion: The Guard clause should be above before click handlers, and other setup is done, so it can skipped if not required.